polynote-kernel/src/main/scala/polynote/kernel/util/KernelContext.scala

```diff
@@ -89,6 +89,35 @@ final case class KernelContext(global: Global, classPath: List[File], classLoade
     stringRepr.toList ++ otherReprs.toList
   }
 
+  @volatile private var currentTaskThread: Thread = _
+  private object taskThreadMonitor
+  private object taskMonitor
+
+  def runInterruptible[T](thunk: => T): T = taskMonitor.synchronized {
+    val prev = taskThreadMonitor.synchronized {
+      val prev = currentTaskThread
+      currentTaskThread = Thread.currentThread()
+      prev
+    }
+
+    val result = try thunk finally {
+      taskThreadMonitor.synchronized {
+        currentTaskThread = prev
+      }
+    }
+
+    result
+  }
+
+  def interrupt(): Unit = {
+    taskThreadMonitor.synchronized(Option(currentTaskThread)) match {
+      case None => ()
+      case Some(thread) => try thread.interrupt() catch {
+        case err: NullPointerException => ()
```

Comment:
When would this occur?

Reply:
Hmm, I was seemingly running into situations where the thread reference would be nulled out from under me. But it turned out to be a symptom of something else, so I can probably remove that.

Reply:
We could keep it and add a log there in case it happens again, maybe?